packages/@aws-cdk/core/lib/custom-resource-provider/cross-region-ssm-reader-handler/index.ts

```diff
@@ -0,0 +1,125 @@
+/*eslint-disable no-console*/
+/* eslint-disable import/no-extraneous-dependencies */
+import { SSM } from 'aws-sdk';
+
+export async function handler(event: AWSLambda.CloudFormationCustomResourceEvent) {
+  const props = event.ResourceProperties;
+  const imports: string[] = props.Imports;
+  const keyName: string = `cdk-strong-ref:${props.StackName}`;
+
+  const ssm = new SSM({ region: props.Region });
+  try {
+    switch (event.RequestType) {
+      case 'Create':
+        console.info('Tagging SSM Parameter imports');
+        await addTags(ssm, imports, keyName);
+        return;
```

Comment:
I was thinking that we want to tag the resource _after_ it is consumed. That way if it is not consumed (maybe the stack operation fails) we don't add the tags. Although I guess in that case we would rollback this resource as well which should remove the tags.

Also, if we read the value through this resource then in order for updates to be consumed then this resource needs to be updated. Right now I guess that is not a problem since we do not allow updates to exported values, but if we introduced weak references in the future that might be a problem.

Reply:
> Although I guess in that case we would rollback this resource as well which should remove the tags.
👆 this
 
If consuming and tagging are not connected, then we might have a window where a parameter is consumed and not tagged, which is dangerous because it might be replaced in that window.

> Also, if we read the value through this resource then in order for updates to be consumed then this resource needs to be updated.

We could solve that by sending the SSM values in as `Properties` -- that way if the value changes, the CR is invalidated.

Reply:
> We could solve that by sending the SSM values in as Properties -- that way if the value changes, the CR is invalidated.

genius!

Reply:
Ok this is updated.